Products/CMFPlone/utils.py

```diff
@@ -234,15 +237,13 @@ def getEmptyTitle(context, translated=True):
 
 
 def typesToList(context):
-    ntp = getToolByName(context, 'portal_properties').navtree_properties
-    ttool = getToolByName(context, 'portal_types')
-    bl = ntp.getProperty('metaTypesNotToList', ())
-    bl_dict = {}
-    for t in bl:
-        bl_dict[t] = 1
-    all_types = ttool.listContentTypes()
-    wl = [t for t in all_types if not t in bl_dict]
-    return wl
+    registry = getUtility(IRegistry)
+    from Products.CMFPlone.interfaces import INavigationSchema
```

Comment:
so this creates a circular import problem in the header.
I looked up why:
in https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/interfaces/controlpanel.py#L4 we import from utils a json validator. this validator is used *only* in ``interfaces/controlpanel``. I propose to either move the validator out of utils into ``plone.app.z3cform`` - we have already validations defined there and json validation is very generic.